src/model/page-config-resolver.js

```diff
@@ -292,6 +335,120 @@ class MicrodataParser {
+      if (!content) {
+        continue;
+      }
+      if (content.toLowerCase() == 'true') {
+        value = true;
+      } else if (content.toLowerCase() == 'false') {
+        value = false;
+      }
+      if (this.isValidElement_(element, root, ALREADY_SEEN)) {
+        return !value;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Verifies if a node is valid based on the following
+   * - child of an item of type 'NewsArticle'
+   * - not a child of an item of type 'Section'
+   * - not seen before, marked using the alreadySeen tag
+   * @param current the node to be verified
+   * @param root the parent to track up to
+   * @param alreadySeen used to tag already visited nodes
+   * @return {!boolean} valid node
+   * @private
+   */
+  isValidElement_(current, root, alreadySeen) {
+    while (current && !current[alreadySeen]) {
+      current[alreadySeen] = true;
+      if (current.hasAttribute('itemscope')) {
+        const type = current.getAttribute('itemtype');
+        if (type == 'http://schema.org/NewsArticle') {
+          // This parent may need to be check for other nodes
+          current[alreadySeen] = false;
+          return true;
+        } else if (type.indexOf('Section') >= 0) {
+          return false;
+        }
+      }
+      current = current.parentNode;
+    }
+    return false;
+  }
+
+  /**
+   * Obtains the product ID that meets the requirements
+   * - child of an item of type 'NewsArticle'
+   * - Not a child of an item of type 'Section'
+   * - child of an item of type 'productID'
+   * @param {Node} root A node that is an item of type 'NewsArticle'
+   * @return {?string} product ID, if found
+   * @private
+   */
+  discoverProductId_(root) {
+    const ALREADY_SEEN = 'alreadySeenForProductInfo';
+    const nodeList = root
+        .querySelectorAll("[itemprop='productID']");
+    for (let i = 0; nodeList[i]; i++) {
+      const element = nodeList[i];
+      const content = element.content || element.textContent;
+      const item = element.closest('[itemtype][itemscope]');
+      const type = item.getAttribute('itemtype');
+      if (type.indexOf('Product') <= -1) {
+        continue;
+      }
+      if (this.isValidElement_(item.parentNode, root, ALREADY_SEEN)) {
+        return content;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Extracts microdata embedded in the DOM
+   * @return {!Microdata} microdata found in the doc
+   */
+  tryExtractConfig_() {
+    let itemId = 0;
+    const results = [];
+    const nodeList = this.doc_.getRootNode().querySelectorAll(
+        "[itemscope][itemtype='http://schema.org/NewsArticle']");
+    const domReady = this.doc_.isReady();
+    for (let i = 0; nodeList[i]; i++) {
+      const element = nodeList[i];
+      if (!domReady && !hasNextNodeInDocumentOrder(element)) {
+        continue;
+      }
+      const locked = this.discoverLocked_(element);
+      const productInfo = this.discoverProductId_(element);
+      const props = {};
```

Comment:
If both are found, why not just return the config right away? What value do we get from storing a set of them? If we want to keep the track of `locked` or `productId` in case one of them already found - we can just store them as private vars in this class. E.g.
```
/** @private {?boolean} */
this.locked_ = null;
```

Then we can avoid discovering it many times, i.e.
```
if (this.locked_ === null) {
  this.locked_ = discoverLocked(...);
}
```

Reply:
Done.